apisix/ssl.lua

```diff
@@ -153,6 +159,9 @@ local function aes_decrypt_pkey(origin, field)
     for _, aes_128_cbc_with_iv in ipairs(aes_128_cbc_with_iv_tbl) do
         local decrypted = aes_128_cbc_with_iv:decrypt(decoded_key)
         if decrypted then
```

Comment:
I prefer this style, clean the error if fail to call `decrypt`.

```lua

local decrypted = aes_128_cbc_with_iv:decrypt(decoded_key)
if not decrypted then
    -- clean the error in OpenSSL
    if C.ERR_peek_error() then
        C.ERR_clear_error()
    end

else
    return decrypted
end

... ...

Reply:
Basically, the current way is only clear of accumulated errors on the one occasion when it finally succeeds, while all failures of decryption cause errors to be thrown to Nginx.

I think this should be acceptable? Maybe it's not necessary to clean up the error every time?

Reply:
Agree.

> Can you confirm this? can it work properly?

Before performing a TLS handshake, nginx will clear the error queue of OpenSSL. It needs to ensure that the error is indeed generated during the TLS handshake, in order to reject the request and not be affected by other unexpected errors